docs/src/main/asciidoc/hibernate-search-standalone-elasticsearch.adoc

```diff
@@ -0,0 +1,1061 @@
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.books = books;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Author)) {
+            return false;
+        }
+
+        Author other = (Author) o;
+
+        return Objects.equals(id, other.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31;
+    }
+}
+----
+
+[source,java]
+----
+package org.acme.hibernate.search.elasticsearch.model;
+
+import java.util.Objects;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToOne;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+
+public class Book {
+
+    public UUID id;
+
+    public String title;
+
+    public Book(UUID id, String title) {
+        this.id = id;
+        this.title = title;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Book)) {
+            return false;
+        }
+
+        Book other = (Book) o;
+
+        return Objects.equals(id, other.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31;
+    }
+}
+----
+
+== Using Hibernate Search annotations
+
+Enabling full text search capabilities for our entities is as simple as adding a few annotations.
+
+Let's edit the `Author` entity to include this content:
+
+[source,java]
+----
+package org.acme.hibernate.search.elasticsearch.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.hibernate.search.engine.backend.types.Sortable;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.IdProjection;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.IndexedEmbedded;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.KeywordField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.ProjectionConstructor;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.SearchEntity;
+
+@SearchEntity // <1>
+@Indexed // <2>
+public class Author {
+
+    @DocumentId // <3>
+    public UUID id;
+
+    @FullTextField(analyzer = "name") // <4>
+    @KeywordField(name = "firstName_sort", sortable = Sortable.YES, normalizer = "sort") // <5>
+    public String firstName;
+
+    @FullTextField(analyzer = "name")
+    @KeywordField(name = "lastName_sort", sortable = Sortable.YES, normalizer = "sort")
+    public String lastName;
```

Comment:
We don't need any kind of proxying, so... It's really up to users. It would work with getters too, it's just that it would be more verbose for no reason.

I will at least mention that in a callback.